cppapi/client/devapi_base.cpp

```diff
@@ -2688,6 +2669,38 @@ DeviceProxy::~DeviceProxy()
 #endif
 }
 
+void DeviceProxy::unsubscribe_all_events()
+{
+    if (ApiUtil* api = ApiUtil::instance())
+    {
+        if (ZmqEventConsumer* zmq_event_consumer = api->get_zmq_event_consumer())
+        {
+            vector<int> event_ids;
+            zmq_event_consumer->get_subscribed_event_ids(this, event_ids);
+
+            for (vector<int>::iterator event_id = event_ids.begin();
+                event_id != event_ids.end();
+                ++event_id)
+            {
+                try
+                {
+                    unsubscribe_event(*event_id);
+                }
+                catch (CORBA::Exception& e)
+                {
+                    Tango::Except::print_exception(e);
+                }
+                catch (...)
+                {
+                    std::cerr <<
+                        "DeviceProxy::unsubscribe_all_events(): "
+                        "Unknown exception thrown from unsubscribe_event() for"
```

Comment:
a space is missing at the end between "for" and "device"

Reply:
Thanks! Fixed.